iosApp/iosApp/ViewModels/ErrorBannerViewModel.swift

```diff
@@ -11,30 +11,39 @@ import shared
 
 class ErrorBannerViewModel: ObservableObject {
     let errorRepository: IErrorBannerStateRepository
+    let settingsRepository: ISettingsRepository
 
     @Published
     private(set) var errorState: ErrorBannerState? = nil
 
     @Published
     var loadingWhenPredictionsStale: Bool
 
+    @Published
+    var showDebugMessages: Bool = false
+
     // option for testing
     var skipListeningForStateChanges = false
 
     init(
         errorRepository: IErrorBannerStateRepository = RepositoryDI().errorBanner,
+        settingsRepository: ISettingsRepository = RepositoryDI().settings,
         initialLoadingWhenPredictionsStale: Bool = false,
+        showDebugMessages: Bool = false,
         skipListeningForStateChanges: Bool = false
     ) {
         self.errorRepository = errorRepository
+        self.settingsRepository = settingsRepository
         loadingWhenPredictionsStale = initialLoadingWhenPredictionsStale
         errorState = self.errorRepository.state.value
+        self.showDebugMessages = showDebugMessages
         self.skipListeningForStateChanges = skipListeningForStateChanges
     }
 
     @MainActor
     func activate() async {
         errorRepository.subscribeToNetworkStatusChanges()
+        showDebugMessages = await (try? settingsRepository.getSettings()[.devDebugMode]?.boolValue) ?? false
```

Comment:
I wish reading settings was synchronous, but the docs [warn against it](https://developer.android.com/topic/libraries/architecture/datastore#synchronous).